app.go

```diff
@@ -179,6 +179,36 @@ func (og optionGroup) String() string {
 	return fmt.Sprintf("fx.Options(%s)", strings.Join(items, ", "))
 }
 
+// OnStartTimeout changes the application's start timeout.
+func OnStartTimeout(v time.Duration) Option {
+	return optionFunc(func(app *App) {
+		app.onStartTimeout = v
+	})
+}
+
+// OnStopTimeout changes the application's start timeout.
+func OnStopTimeout(v time.Duration) Option {
+	return optionFunc(func(app *App) {
+		app.onStopTimeout = v
+	})
+}
+
+// GetStartTimeout returns on start timeout.
+//
+// Most users won't need to use this method, because they may set timeout if necessary.
+// This can be useful for unit tests when need to read it to pass then to Start method.
+func (app *App) GetStartTimeout() time.Duration {
+	return app.onStartTimeout
+}
+
+// GetStopTimeout returns on start timeout.
+//
+// Most users won't need to use this method, because they may set timeout if necessary.
+// This can be useful for unit tests when need to read it to pass then to Stop method.
+func (app *App) GetStopTimeout() time.Duration {
+	return app.onStopTimeout
+}
```

Comment:
I'd prefer to remove these. They don't help users write meaningful tests - we're basically calling `SetFoo` and testing by calling `GetFoo`. Instead, we should test by setting the start timeout and verifying that we can actually block for however long is necessary.

Reply:
Ah, I see - we need to export *some* way to get this information from `fxtest`.